jenkins/files/jobs/_common.xml

```diff
@@ -50,10 +50,10 @@
               <compareType>{{ project.compare_type|default("PLAIN") }}</compareType>
               <pattern>{{ project.name|default(pname) }}</pattern>
               <branches>
-                {%- for branch in project.branches %}
+                {%- for bname, branch in project.branches %}
```

Comment:
This will not work, you cannot extract key and value from a list
You should determine whether branches is list or dict (mapping) first